### PR TITLE
Update deprecated repository methods usages

### PR DESCRIPTION
**Pull request by johnnyomair**
Use the underlying entity manager instead.

### CODE REVIEW

File: api/src/documents/links/links.resolver.ts [hunk 1]
```diff
@@ -60,14 +60,14 @@ export class LinksResolver {
                 content: input.content.transformToBlockData(),
             });
 
-            this.repository.persist(link);
+            this.repository.getEntityManager().persist(link);
```
Comment (johnnyomair) [inline api/src/documents/links/links.resolver.ts:63]: What do we prefer?

1. Inject entity manager and call `this.entityManager.persist()` (the API Generator does this)
2. Use `repository.getEntityManager()`

Comment (johnnyomair) [reply api/src/documents/links/links.resolver.ts:63]: Probably 1️⃣ , but I don't have a strong opinion on this. I've seen both in the core repo.

Comment (johnnyomair) [reply api/src/documents/links/links.resolver.ts]: Changed: 2e2312f80899b0345ef0581501292ba9d207c73f

**Comment by johnnyomair** [comment]
I've made this a recommended change because it will be required when upgrading to MikroORM 6 / Comet 8.

**Review by nsams — APPROVED** [review]

**Review by dkarnutsch — APPROVED** [review]

**Pull request merged**